pkg/controller/direct/compute/firewallpolicyrule_controller.go

```diff
@@ -192,9 +192,11 @@ func (a *firewallPolicyRuleAdapter) Update(ctx context.Context, updateOp *direct
 	if mapCtx.Err() != nil {
 		return mapCtx.Err()
 	}
-	// The field priority should be removed from the patch request body and included as a query parameter.
-	// See API doc: https://cloud.google.com/compute/docs/reference/rest/v1/firewallPolicies/patchRule#query-parameters
-	firewallPolicyRule.Priority = nil
+
+	// Assign API output-only values
+	// todo: https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/4455
+	firewallPolicyRule.Kind = a.actual.Kind
+	firewallPolicyRule.RuleTupleCount = a.actual.RuleTupleCount
 
```

Comment:
Assigning `Kind` and `RuleTupleCount` here effectively prevents `CompareProtoMessage` from flagging them as diffs, which is necessary.

However, since `firewallPolicyRule` is later used as the request body, these output-only fields will now be sent to the API. While the PR description confirms this works with real GCP (and the fields are likely ignored), you might consider nil-ing them out again *after* the `CompareProtoMessage` check (similar to how you handle `Priority` below) if you prefer to keep the request body clean.